### ISSUE TITLE
[Accessibility] WCAG 2.1 1.3.1, and 4.1.2 && Section 508 502.3.1 && Screen Reader Usability

### ISSUE THREAD

**Issue by KimPlausible** [swagger-api/swagger-ui#5248]
### Q&A (please complete the following information)
 - OS: Windows
 - Browser: Microsoft Edge
 - Version: 42.17134.1.0​​
 - Method of installation: npm
 - Swagger-UI version: 3.17.6
 - Swagger/OpenAPI version: latest

### Describe the bug you're encountering

Aggregated Accessibility Bugs related to WCAG 2.0 1.3.1 && Screen Reader Usability
1. Narrator (screen reader) does not read the value against the Value label which is represented in the form of asterisks using Caps Lock arrow keys​. Screen reader users will not be able to understand the context of the asterisks because it's not read by the screen reader.​
2. Name of the combo box "Schemes" is showing as "List items present inside the combo box".
3. Narrator (screen reader) does not read the reference of column header (i.e. code details ,code and description). The user will not be able to understand the context of each column.
4. Narrator (screen reader) reads the required field as asterisks only (id)(Present inside the delete which is part of Report section). The user will not understand that the field is required.
5. Narrator (screen reader) does not read the "Error message" when we activate "Execute button". It reads the error message when we again move to the "Id field". The user will not be able to get the information about the error message within context.
6. The name property of a focusable element must not be null.
7. Name of combo box is showing as list items for Parameter content type.
8. Narrator (screen reader) reads the state of the column as "Non selected", which is not required. The user will get confused when Narrator (screen reader) reads the unnecessary information.
9. Narrator (screen reader) does not read the "Expand/Collapsed Status of the link (PUT/pet) ". The user will not be able to get the expand/collapsed status.
10. Name of the group is missing.
11. When Narrator (screen reader) users navigates within the table present under the "Responses" section, after reading the first row below header it announces "Example" and "Model" links. As a result, the screen reader users will confused and not able to understand the functionality effectively.
12. The accordions provided on the page such as, "pet" etc. expand the content below upon activation. But for the same purpose, a button is also provided for expand/collapse labeled as "Expand/Collapse operation". If two different elements are provided for same functionality as discussed above, screen reader will be confused about the functionality and fail to use it.
13. On the page, when screen reader users activates "Try it out", "Execute" buttons, the content gets dynamically changes. Regarding these no instruction or alert is provided. As a result, screen reader users not able to understand the same and fails to use it.
14. The button provided on the page "Collapse operation"  is markup using H4 which is inaccurate. Screen reader users navigates with navigation quick keys such as "B" for Button and "H" for Heading etc.; if multiple elements are used for marking a controls as discussed, then screen reader users will confused about the same and fails to understand.
15. When screen reader users inputs value in the value text-field and activates "Authorize" button, the dialog content gets dynamically changes above. But no alert/feedback is announced for screen reader users. If no alert/feedback is announced, then screen reader users will not able to understand the same and fail to use it.
16. When screen reader users activates the "Authorize " button, a dialog gets displayed on the screen which is not announced. Also, the screen reader users focus moves out of the dialog in the background. If no alert dialog is provided, screen reader users won't be aware about the presence of the dialog. Moreover, if the focus moves out of the dialog then, they will not able to distinguished between the background content and the dialog content.
17. 'Value' form element missing label
18. 'X' button on popup does not have Name
19. Editable box for 'Value' does not have Name
20. Name of "Parameters" and "Responses" tables missing
21. Name property of 'Curl' is null
22. Group (little expand carrot in subsection under 'Models') doesn't have Name
23. Close modal button does not have discernable text
24. reportPayload's Edit Value box missing Name
25. Name of the cell is missing.

### To reproduce...
Steps to reproduce the behavior
1. Ctrl+Win+Enter to start Narrator in Windows -> https://petstore.swagger.io/ -> Authorize -> enter Value for api_key -> Authorize -> verify Value for updated field via screenreader
2. https://petstore.swagger.io/ -> inspect element: combo box "HTTPS"
3. https://petstore.swagger.io/ -> navigate to Response table for GET/pet/findByStatus -> Ctrl+Win+Enter to start Narrator in Windows -> Use Caps+Arrow keys to read out references of column headers
4. https://petstore.swagger.io/#/pet/deletePet -> Ctrl+Win+Enter to start Narrator in Windows -> Use Caps+Arrow keys to read out the *required field
5. Ctrl+Win+Enter to start Narrator in Windows -> https://petstore.swagger.io/#/user/deleteUser -> Try it out -> leave username field blank -> Execute -> navigate to username field for error message
6. https://petstore.swagger.io/#/user/getUserByName -> Try it out -> input user1 in username field, inspect element: curl body
7. https://petstore.swagger.io/#/pet/addPet -> inspect element: combo box "application/json-patch+json"
8. Ctrl+Win+Enter to start Narrator in Windows -> https://petstore.swagger.io/#/pet/findPetsByStatus ->navigate to Parameters table -> Use Caps+Arrow keys to read out table values
9. Ctrl+Win+Enter to start Narrator in Windows -> https://petstore.swagger.io/#/pet/updatePet -> expand and collapse PUT/pet operation
10. https://petstore.swagger.io/#/ -> Models -> inspect element: group ""
11. https://petstore.swagger.io/#/pet/findPetsByStatus -> Ctrl+Win+Enter to start Narrator in Windows -> navigate to Responses table -> Use Caps+Arrow keys to read Row2,Column2
12. https://petstore.swagger.io/#/pet -> pet -> Expand operation
13. Ctrl+Win+Enter to start Narrator in Windows -> https://petstore.swagger.io/#/store/getInventory -> 
14. https://petstore.swagger.io/#/ -> Inspect element: Expand/Collapse button
15. Ctrl+Win+Enter to start Narrator in Windows -> https://petstore.swagger.io/#/ -> Authorize -> enter value for api_key -> Authorize
16. Ctrl+Win+Enter to start Narrator in Windows -> https://petstore.swagger.io/#/ -> Authorize
17. https://petstore.swagger.io/#/store/getInventory -> Available authorizations -> inspect Value edit box
18. https://petstore.swagger.io/ -> Authorize -> inspect 'X' button for Close
19. https://petstore.swagger.io/#/store/getInventory -> Available authorizations -> inspect Value edit box
20. https://petstore.swagger.io/#/pet/findPetsByStatus -> Inspect "Parameters" & "Responses" tables
21. https://petstore.swagger.io/#/pet/findPetsByStatus -> Try it out -> Execute -> inspect Curl
22. https://petstore.swagger.io/#/ -> Models -> Order -> inspect Expand/Collapse carrot
23. https://petstore.swagger.io/#/-> Authorize -> inspect 'X' button for Close
24. https://petstore.swagger.io/#/pet/addPet -> Try it out -> Edit Value
25. https://petstore.swagger.io/#/pet/deletePet -> inspect apiKey Description cell

### Expected behavior

1. Narrator (screen reader) should read the value against the Value label, which is represented in the form of asterisks, and should inform the user that the value is encoded and secured.​
2. Name should be provided as "Schemes".
3. Narrator (screen reader) should read the reference of column header (i.e. code and description).
4. Narrator (screen reader) should read the mandatory field as required field.
5. Narrator (screen reader) should read the error message when we activate "Execute" button.
6. The name property of a focusable element must not be null.
7. Name should be provided for combo box Parameter content type.
8. Narrator (screen reader) should not read the state of the column as "Non selected".
9. Narrator (screen reader) should read out the Expand/Collapsed state.
10. Proper name should be provided for group.
11. Table content should be announced accurately and in the structured manner for the benefits of screen reader users irrespective of screen reader used.
12. Merge both link and button discussed above in single element for the benefits of screen reader users.
13. Provide a hidden instruction about the dynamically updated contents for the benefits of screen reader users.
14. The button should be removed from the H4 tag in the source code for the benefits of screen reader users.
15. An alert or feedback should be announced when user activates the discussed button for the benefits of screen reader users.
16. "Dialog" should be announced and popup start and popup end should be provided at the starting and ending of the dialog; alternatively, trap the user focus within the dialog.
17. Form elements must have labels.
18. Name should be provided as 'Close'.
19. Name should be provided as 'Value'.
20. Names should be provided for both tables as "Parameters" and "Responses", respectively.
21. The name property of focusable element 'Curl' must not be null.
22. Proper name should be provided for a group.
23. Buttons must have discernable text.
24. The name property of a focusable element must not be null.

### Screenshots

1. 
![image](https://user-images.githubusercontent.com/12721313/54640834-f1bad000-4a4d-11e9-9573-cd0f6fe640dc.png)
2. 
![image](https://user-images.githubusercontent.com/12721313/54641210-da301700-4a4e-11e9-868b-98a60d343589.png)
3. 
![image](https://user-images.githubusercontent.com/12721313/54641504-9689dd00-4a4f-11e9-85e1-cf5a27938d9d.png)
4. 
![image](https://user-images.githubusercontent.com/12721313/54645413-2a13db80-4a59-11e9-9ccb-b5c6cf0a08bc.png)
5. 
![image](https://user-images.githubusercontent.com/12721313/54646765-6fd2a300-4a5d-11e9-96c4-d486ca9099ac.png)
6. 
![image](https://user-images.githubusercontent.com/12721313/54647053-5f6ef800-4a5e-11e9-8e84-d9d099b67b20.png)
7. 
![image](https://user-images.githubusercontent.com/12721313/54647859-63e8e000-4a61-11e9-8b04-10dc57f875d1.png)
8. 
![image](https://user-images.githubusercontent.com/12721313/54648273-dad2a880-4a62-11e9-8e50-f4fe30230c26.png)
9. 
![image](https://user-images.githubusercontent.com/12721313/54648629-45381880-4a64-11e9-8bca-896ec59ecac7.png)
10. 
![image](https://user-images.githubusercontent.com/12721313/54648885-25edbb00-4a65-11e9-97b7-0e12a6104704.png)
11. 
![image](https://user-images.githubusercontent.com/12721313/54649705-8f22fd80-4a68-11e9-9bd3-29f05f180d55.png)
12. 
![image](https://user-images.githubusercontent.com/12721313/54650594-86ccc180-4a6c-11e9-81d6-07d3b6d2a412.png)
13. 
![image](https://user-images.githubusercontent.com/12721313/54651278-618d8280-4a6f-11e9-8ac4-548b78cdd267.png)
14. 
![image](https://user-images.githubusercontent.com/12721313/54651629-f8a70a00-4a70-11e9-9954-998e7dc62b6b.png)
15. 
![image](https://user-images.githubusercontent.com/12721313/54651822-bd590b00-4a71-11e9-9afa-4f5b42b06dba.png)
16. 
![image](https://user-images.githubusercontent.com/12721313/54651995-6a338800-4a72-11e9-9ca1-de3073cf1a74.png)
17. 
![image](https://user-images.githubusercontent.com/12721313/55837677-ee5bb700-5ad6-11e9-9226-64629da63da2.png)
18. 
![image](https://user-images.githubusercontent.com/12721313/55836794-64aaea00-5ad4-11e9-9fad-0f0e3eab3b3a.png)
19. 
![image](https://user-images.githubusercontent.com/12721313/55837474-5958be00-5ad6-11e9-8bb1-57067a4c8fb1.png)
20. 
![image](https://user-images.githubusercontent.com/12721313/55838103-27e0f200-5ad8-11e9-8708-93baa427242c.png)
21. 
![image](https://user-images.githubusercontent.com/12721313/55838364-11876600-5ad9-11e9-9613-ab0fb38e907c.png)
22. 
![image](https://user-images.githubusercontent.com/12721313/55838639-fb2dda00-5ad9-11e9-81fd-e83b5653b90c.png)
23. 
![image](https://user-images.githubusercontent.com/12721313/55895308-c325a600-5b70-11e9-8669-3ae52d0857e7.png)
24. 
![image](https://user-images.githubusercontent.com/12721313/55895715-93c36900-5b71-11e9-845b-f7a44e8b1492.png)
25. 
![image](https://user-images.githubusercontent.com/12721313/57406110-dee29300-7194-11e9-84e0-f376e358e832.png)

**Comment by linelson** [comment]
@shockey would you be open to a PR to fix these accessibility bugs as well as https://github.com/swagger-api/swagger-ui/issues/5247?

**Comment by KimPlausible** [comment]
**Pri 0** (Blocking, high impact): 1, 4, 5, 7, 9, 13, 15, 16
**Pri 1** (Blocking, low impact): 2, 11
**Pri 2** (Not blocking, high impact): 3, 6, 10, 14
**Pri 3** (Not blocking, low impact): 8, 12

**Comment by msftedad** [comment]
Hi @KimPlausible, is there any update on this bug, and could you please provide a tentative ETA for fixing it?

**Comment by msftedad** [comment]
Hi @ponelat, is there any update on this bug and could you please provide a tentative ETA for fixing it.

**Comment by msftedad** [comment]
Hi @maggiewachs, this issue is still repro'ing in the latest version of Swagger UI(5.20.1). Could you please confirm if the timeline for fixing it has been finalized? Thanks

CC: @ponelat

**Comment by msftedad** [comment]
Hi Team, Any update on this issue?
